File: cpp-icestorm/publisher.cpp
#include <Ice/Application.h>
#include <IceStorm/IceStorm.h>
#include "Printer.h"

using namespace std;
using namespace Ice;
using namespace IceStorm;
using namespace Example;

class Publisher : public Application {
  TopicManagerPrx get_topic_manager() {
    string key = "IceStorm.TopicManager.Proxy";
    ObjectPrx base = communicator()->propertyToProxy(key);
    if (!base) {
      cerr << "property " << key << " not set." << endl;
      return 0;
    }

    cout << "Using IceStorm in: '" << key << "' " << endl;
    return TopicManagerPrx::checkedCast(base);
  }

public:
  virtual int run(int argc, char*[]) {
    TopicManagerPrx topic_mgr = get_topic_manager();
    if (!topic_mgr) {
      cerr << appName() << ": invalid proxy" << endl;
      return EXIT_FAILURE;
    }

    string topic_name = "PrinterTopic";
    TopicPrx topic;
    try {
      topic = topic_mgr->retrieve(topic_name);
    } catch (const NoSuchTopic& e) {
      cerr << appName()
	   << ": no sucho topic found, creating" << endl;
      topic = topic_mgr->create(topic_name);
    }

    assert(topic);

    ObjectPrx publisher = topic->getPublisher();
    PrinterPrx printer = PrinterPrx::uncheckedCast(publisher);

    cout << "publishing 10 'Hello World' events" << endl;
    for(char i='0'; i <= '9'; ++i)
      printer->write(string("Hello World ") + i + "!");

    return EXIT_SUCCESS;
  }
};

int main(int argc, char* argv[]) {
  Publisher app;
  return app.main(argc, argv);
}

File: java-icestorm/Publisher.java
import Ice.*;
import IceStorm.*;
import Example.*;

public class Publisher extends Application {
  public int run(String[] args) {
    String key = "IceStorm.TopicManager.Proxy";
    ObjectPrx prx = communicator().propertyToProxy(key);
    TopicManagerPrx manager = TopicManagerPrxHelper.checkedCast(prx);

    if (manager == null) {
      System.err.println("invalid proxy");
      return 1;
    }

    String topicName = "PrinterTopic";
    TopicPrx topic;
    try {
      topic = manager.retrieve(topicName);
    } catch(NoSuchTopic e) {
      try {
        topic = manager.create(topicName);
      } catch(TopicExists ex) {
        System.err.println("Temporary failure, try again.");
        return 1;
      }
    }

    ObjectPrx publisher = topic.getPublisher();
    PrinterPrx printer = PrinterPrxHelper.uncheckedCast(publisher);

    System.out.println("publishing 10 'Hello World' events");
    for(int i=0; i < 10; i++)
       printer.write(String.format("Hello World %s!", i));

    return 0;
  }

  public static void main(String[] args) {
    Publisher app = new Publisher();
    int status = app.main("Publisher", args);
    System.exit(status);
  }
}
